### PR TITLE
refactor: handle transaction filter types

### PR DESCRIPTION
**Pull request by alexbarnsley**
## Summary

https://app.clickup.com/t/86dv5z208

TODO:

- [x] Add filter options for unvote, validator registration & validator resignation
- [x] tests?
- [x] phpstan?
- [x] ~Add a card to move away from enum for payload hashes - go along the route of the Network facade (maybe even use the same class - `app/Services/Network.php`)~ - https://app.clickup.com/t/86dv95n20

Also added a card to look at the speed of the transaction table filters - https://app.clickup.com/t/86dv9ee09

## Checklist

-   [ ] I checked my UI changes against the design and there are no notable differences
-   [ ] I checked my UI changes for any responsiveness issues
-   [ ] I checked my UI changes in light AND dark mode
-   [ ] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I opened a corresponding card on Clickup for any remaining TODOs in my code
-   [ ] I added a short description on how to test this PR _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

File: app/Models/Transaction.php [hunk 1]
```diff
@@ -194,6 +200,56 @@ public function recipient(): Wallet
...
+                    });
+                    // ->orWhere(fn ($query) => $query->when($filter['others'] === true, fn ($query) => $query
+                        //     ->orWhere(
+                        //         fn ($query) => $query
+                        //             ->whereNotIn('type', [
+                        //                 TransactionTypeEnum::TRANSFER,
+                        //                 TransactionTypeEnum::VOTE,
+                        //                 TransactionTypeEnum::MULTI_PAYMENT,
+                        //             ])
+                        //     )));
```
Comment (ItsANameToo) [inline app/Models/Transaction.php:248]: remove?

File: config/arkscan.php [hunk 2]
```diff
@@ -55,7 +55,9 @@
             'blockReward'         => intval(env('ARKSCAN_NETWORK_BLOCK_REWARD', 2)),
             'base58Prefix'        => intval(env('ARKSCAN_NETWORK_BASE58_PREFIX', 30)),
 
-            'contract_addresses' => [],
+            'contract_addresses' => [
+                'consensus' => env('ARKSCAN_CONSENSUS_CONTRACT_ADDRESS', '0x535B3D7A252fa034Ed71F0C53ec0C6F784cB64E1'),
```
Comment (ItsANameToo) [inline config/arkscan.php:59]: will this data work with the `isContract` logic that expects an array of addresses instead?

**Review by ItsANameToo — CHANGES_REQUESTED** [review]

File: config/arkscan.php (hunk 2 above)
Comment (alexbarnsley) [reply config/arkscan.php:59]: It should yeh but I'll add a test to the MemoryWallet to be sure

File: app/Models/Transaction.php (hunk 1 above)
Comment (alexbarnsley) [reply app/Models/Transaction.php:248]: ah yes

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**